lib/completion.dart

```diff
@@ -67,11 +67,28 @@ class DartCompleter extends CodeCompleter {
                   SourceEdit(edit.length, edit.offset, edit.replacement))
               .toList();
 
+          int absoluteCursorPosition;
+
+          // TODO(redbrogdon): Find a way to properly use these linked edit
+          // groups via selections and multiple cursors.
+          if (assist.linkedEditGroups != null &&
+              assist.linkedEditGroups.isNotEmpty) {
+            absoluteCursorPosition =
+                assist.linkedEditGroups.first.positions?.first?.offset;
```

Comment:
does this work with test code? Are `test.dart` and `main.dart` concatenated when they are analyzed? Or does the analyzer treat them as separate files?

Reply:
The analyzer receives it all as one big file:

```dart
      final fullSource = '$userSource\n'
          '${context.testMethod}\n'
          '${executionSvc.testResultDecoration}\n';
```

This is the same strategy we use for compilation so that line numbers are maintained as the user expects them.